Fixes #455 - Added documentation for modifying message replay mechanism

cha-ku opened the pull request:
Added the documentation for changing replay mechanism in subscribing.rst

cha-ku committed:
Fixes #455 - Added documentation for modifying message replay mechanism

jeremycline commented on doc/deployment.rst:
```diff
@@ -213,15 +213,14 @@ the tables.  It will read in that connection url from
-messages, and store them all in the db.
-
-::
+messages, and store them all in the db.::
```
The changes in this file aren't related to the rest of the commit and don't seem necessary to me. Please don't include them in this commit. Thanks!

jeremycline commented on doc/subscribing.rst:
```diff
@@ -47,6 +47,17 @@ Consuming Non-ZeroMQ Messages
+missed from the datagrepper service. To disable this functionality:
+
+#. Stop fedmsg-hub
+#. Remove the file at on disk at ``/var/run/fedmsg/status/fedmsg-hub/*``
```
This should be "Remove the file on disk at" rather than "Remove the file at on disk at"

jeremycline reviewed (changes requested):
Two small issues, but it's looking good.

Please don't close the pull request as it loses the review history when you do that. Just push a new commit or amend your current commit and force push (preferred).

Thanks!

jeremycline committed:
Release notes for fedmsg-1.1.0

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

cha-ku committed:
Merge branch 'develop' of https://github.com/fedora-infra/fedmsg into develop

Merging changes with the current branch

cha-ku committed:
Issue #455 - Documentation fix for fedmsg-hub command

cha-ku committed:
Removed unnecessary change in line spacing

cha-ku commented:
@jeremycline Thanks for the review! I have made the necessary changes. I think this should be fine now. Please let me know if there are any issues. Thanks again!

pypingou commented:
There seems to be a merge commit in there, maybe try using git rebase ?

jeremycline committed:
Revert "Add a skeleton for the click-based CLI"

This reverts commit 2a6148af87db711eb73a16eb564ee7f698c590d7.

Since the CLI is not currently hooked up to anything, revert this commit
before releasing v1.1.0.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Stop using ``ca_cert_cache`` in the consumer tests

With the recent change to how CA certificates are loaded, just use the
path to certificate.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

sijis committed:
Fix default type check (#500)

* fix: Use int validator for ca and crl cache expiry

* fix: Give expiry keys sane defaults

* test: Update expiry defaults

* fix: Update relay_inbound references to use str

* fix: Set crl_cache_expiry to a better default

ralphbean committed:
Return any return value from ._consume(..).

In a non-zeromq case, it matters what the return value of _consume is.  (It is
a boolean, indicating if the message was successfully handled or not).  This,
in turn, gets used to figure out whether or not to send an ACK back to the
broker.

Here, the return values were being swallowed and turned into `None` values.

ralphbean committed:
Docstring, for a method wanting.

ralphbean committed:
Tests for that return keyword.

jeremycline committed:
Add release notes for v1.1.1

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Bump the release for v1.1.1

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

glensc commented:
the merge commit is there because developer didn't create feature branch, thus this PR is linked to his `develop` branch.

glensc commented:
i think it's simpler for maintainers to cherry pick and fixup the commit and be done with it.

pypingou committed:
Escape the backslash that escapes the dot

This will make python 3.6+ happier and stop sending DeprecationWarning

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

jeremycline committed:
Move configuration documentation into the Python module

This moves the documentation for each configuration value into the
module docblock and alters the Sphinx auto-generated documentation to
include the module-level documentation in the user-facing configuration
page. The configuration API docs used to be the place where
configuration load order was documented and users weren't finding it.

The documentation itself is not changed.

Fixes #505

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Note config files are loaded lexicographical order

Fixes #511

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

cha-ku commented:
@jeremycline @glensc @pypingou Do I need to rebase my changes on top of current develop branch? Or would it be easier to open a new PR?

pypingou commented:
Rebasing and force-pushing to your fork should be straight forward, but the new PR road is always an option

cha-ku commented:
I did a git rebase and then pushed the fork on top. Hope this is alright.
